govcd/ip_space_uplink.go

```diff
@@ -0,0 +1,186 @@
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	typeResponses := []*types.IpSpaceUplink{{}}
+	err = client.OpenApiGetAllItems(apiVersion, urlRef, queryparams, &typeResponses, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wrap all typeResponses into DefinedEntityType types with client
+	results := make([]*IpSpaceUplink, len(typeResponses))
+	for sliceIndex := range typeResponses {
+		results[sliceIndex] = &IpSpaceUplink{
+			IpSpaceUplink: typeResponses[sliceIndex],
+			vcdClient:     vcdClient,
+		}
+	}
+
+	return results, nil
+}
+
+// GetIpSpaceUplinkByName retrieves a single IP Space Uplink by Name in a given External Network
+func (vcdClient *VCDClient) GetIpSpaceUplinkByName(externalNetworkId, name string) (*IpSpaceUplink, error) {
+	queryParams := queryParameterFilterAnd(fmt.Sprintf("name==%s", name), nil)
+	allIpSpaceUplinks, err := vcdClient.GetAllIpSpaceUplinks(externalNetworkId, queryParams)
+	if err != nil {
+		return nil, fmt.Errorf("error getting IP Space Uplink by Name '%s':%s", name, err)
+	}
+
+	return oneOrError("name", name, allIpSpaceUplinks)
+}
+
+// GetIpSpaceUplinkById retrieves IP Space Uplink with a given ID
+func (vcdClient *VCDClient) GetIpSpaceUplinkById(id string) (*IpSpaceUplink, error) {
+	if id == "" {
+		return nil, fmt.Errorf("IP Space Uplink lookup requires ID")
+	}
+
+	client := vcdClient.Client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointIpSpaceUplinks
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint, id)
+	if err != nil {
+		return nil, err
+	}
+
+	response := &IpSpaceUplink{
+		vcdClient:     vcdClient,
+		IpSpaceUplink: &types.IpSpaceUplink{},
+	}
+
+	err = client.OpenApiGetItem(apiVersion, urlRef, nil, response.IpSpaceUplink, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+// Update updates IP Space Uplink
+func (ipSpaceUplink *IpSpaceUplink) Update(ipSpaceUplinkConfig *types.IpSpaceUplink) (*IpSpaceUplink, error) {
+	client := ipSpaceUplink.vcdClient.Client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointIpSpaceUplinks
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	ipSpaceUplinkConfig.ID = ipSpaceUplink.IpSpaceUplink.ID
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint, ipSpaceUplinkConfig.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &IpSpaceUplink{
+		IpSpaceUplink: &types.IpSpaceUplink{},
+		vcdClient:     ipSpaceUplink.vcdClient,
+	}
+
+	err = client.OpenApiPutItem(apiVersion, urlRef, nil, ipSpaceUplinkConfig, result.IpSpaceUplink, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error updating IP Space: %s", err)
+	}
+
+	return result, nil
+}
+
+// Delete deletes IP Space Uplink
```

Comment:
Hm :) "Delete deletes" - have we always used this "double-word" form?

Reply:
Changed as per suggestion https://github.com/vmware/go-vcloud-director/pull/579#discussion_r1243557652